langchain_tools_demo/agent.py

```diff
@@ -91,23 +109,23 @@ async def handle_error_response(response):
         return f"Error sending {response.method} request to {str(response.url)}): {await response.text()}"
 
 
-async def create_client_session() -> aiohttp.ClientSession:
+async def create_client_session(user_id_token: Optional[str]) -> aiohttp.ClientSession:
     return aiohttp.ClientSession(
         connector=await get_connector(),
-        headers=get_header(),
+        headers=get_header(user_id_token),
```

Comment:
Do we need to refresh the user_id_token when it expires? If so, how is that handled?

Reply:
Both the ID tokens ("Authorization" field for cloud run and "User-Id-Token" for user authentication) expire after one hour. I could add a refresher function to re-acquire the headers once every hour in a separate PR maybe?

Reply:
Does the session support dynamic credentials? If not we probably need to switch to grabbing the credentials on the fly